qlib/rl/order_execution/interpreter.py

```diff
@@ -57,18 +57,19 @@ class FullHistoryStateInterpreter(StateInterpreter[SAOEState, FullHistoryObs]):
 
     Parameters
     ----------
-    data_dir
-        Path to load data after feature engineering.
     max_step
         Total number of steps (an upper-bound estimation). For example, 390min / 30min-per-step = 13 steps.
     data_ticks
         Equal to the total number of records. For example, in SAOE per minute,
         the total ticks is the length of day in minutes.
     data_dim
         Number of dimensions in data.
+    data_dir
+        Path to load data after feature engineering. It is optional since in some cases we do not need explicit
+        path to load data. For example, the data has already been preloaded in `init_qlib()`.
     """
 
-    def __init__(self, data_dir: Path, max_step: int, data_ticks: int, data_dim: int) -> None:
+    def __init__(self, max_step: int, data_ticks: int, data_dim: int, data_dir: Path = None) -> None:
```

Comment:
I think all implementations related to `data_dir` is coupled with the specific data format for that specific case.
So it should not be a general class as a part of the framework.
So it should be redesigned after the data interface is well-designed.

We could leave a TODO here in this PR.

Reply:
TODO left.